Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -6,24 +6,14 @@ class UIKitUserInteractions {
 
     private static let lock = NSLock()
 
-    private static let initializeSwizzle: () = {
-        swizzleSendAction()
+    private static let addNotificationObservers: () = {
+        NotificationCenter.default.addObserver(UIKitUserInteractions.self, selector: #selector(amp_didBeginEditing), name: UITextField.textDidBeginEditingNotification, object: nil)
+        NotificationCenter.default.addObserver(UIKitUserInteractions.self, selector: #selector(amp_didEndEditing), name: UITextField.textDidEndEditingNotification, object: nil)
+        NotificationCenter.default.addObserver(UIKitUserInteractions.self, selector: #selector(amp_didBeginEditing), name: UITextView.textDidBeginEditingNotification, object: nil)
+        NotificationCenter.default.addObserver(UIKitUserInteractions.self, selector: #selector(amp_didEndEditing), name: UITextView.textDidEndEditingNotification, object: nil)
```

Comment:
Note that the prefix is intended to allow for name spacing, as objc has a single global namespace. But you only need to do that when you're extending public objects, internal objects are functionally their own namespace. (Ie, we'd have AMPUtils.foo() and UIView.amp_foo(), we just don't need prefix swift only classes because swift has name spacing.)

Reply:
So if I understand correctly, `amp_` is no longer required for the notification handlers since they are defined within an internal class (in this case, the `UIKitUserInteractions` class has its own namespace, thus separating the objc declaration within `UIKitUserInteractions` from the global namespace).

Reply:
Mostly yes, in this case, I'm not sure UIKitUserInteractions is available in objc at all since it's not marked objc and doesn't inherit from NSObject (if it was, it should have a prefix).  Best rule of thumb is to just assume anything you are adding to objc should have be prefixed at the highest level possible, so class / protocol level for most things, and for functions / variables in extensions.